### PR TITLE
in_memory_engine: add a debug API for IME to show current cached regions metadata

### PR DESCRIPTION
**Pull request by glorv**
### What is changed and how it works?

Issue Number: ref #16141

What's Changed:

```commit-message
Add a "/cached_regions" http inteface to fetch current cached regions metadata. 
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: src/server/status_server/mod.rs [hunk 1]
```diff
@@ -734,6 +741,7 @@ where
                                 Self::handle_resume_grpc(grpc_service_mgr)
                             }
                             (Method::GET, "/async_tasks") => Self::dump_async_trace(),
+                            (Method::GET, "/cached_regions") => Self::handle_dumple_cached_regions(in_memory_engine.as_ref()),
```
Comment (overvenus) [inline src/server/status_server/mod.rs:744]: Would you mind placing it in `debug/ime/cached_regions`? I'm thinking of adding a `debug/ime/dump_region` API to dump the metadata and data of a specific cached region. I think it's clearer to put them under the `debug/ime` prefix to indicate their scope.

File: Cargo.toml [hunk 2]
```diff
@@ -486,8 +486,8 @@ overflow-checks = false
 rpath = false
 
 [profile.release.package.server]
-opt-level = 1
-codegen-units = 4
+opt-level = 4
+codegen-units = 1
```
Comment (overvenus) [inline Cargo.toml:490]: I'm curious if this change has any benefits.

**Review by overvenus — COMMENTED** [review]
Rest LGTM

File: Cargo.toml (hunk 2 above)
Comment (glorv) [reply Cargo.toml:490]: This must be a misconfiguration, I'll roll it back.

File: src/server/status_server/mod.rs (hunk 1 above)
Comment (glorv) [reply src/server/status_server/mod.rs]: Done.

**Review by overvenus — APPROVED** [review]

**Review by SpadeA-Tang — APPROVED** [review]

**Pull request merged**